### PR TITLE
TM-5587: Manage EL Position Filtering

### PR DESCRIPTION
**Pull request by pveland621**
ACs:
1. Verify `el_postions_req_mapping` JSON output is valid for each filter (can paste into DEV1 Swagger)

Dual Merge: 
- [FE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2901)

[Ticket](https://metaphase.atlassian.net/browse/TM-5587)

### CODE REVIEW

File: talentmap_api/fsbid/services/positions.py [hunk 1]
```diff
@@ -325,10 +325,42 @@ def get_el_positions(request, jwt_token):
...
+        elif key == 'el-jobs':
+            for jc in request[key].split(','):
+                values_formatted.append(f"{{\"JC_ID\": \"{jc}\"}}")
+            result['PTYP_JC_DD_TAB_I'] = f"{{\"Data\": [{','.join(values_formatted)}]}}"
+        elif key == 'el-language':
+            for lang in request[key].split(','):
+                values_formatted.append(f"{{\"LANG_CODE\": \"{lang}\"}}")
+            result['PTYP_LANGUAGE_TAB_I'] = f"{{\"Data\": [{','.join(values_formatted)}]}}"
+        
+    return result
```
Comment (M00staff) [inline talentmap_api/fsbid/services/positions.py:363]: this looks a lot like the request structure for the `post_access` data - we made a common function that i THINK could have been used here called `format_request_data_to_string` 

not worth holding this up, as this looks to be working just fine, as is ... but, if you have any problems with it down the line or for next time - might be worth looking into

Comment (M00staff) [reply talentmap_api/fsbid/services/positions.py:363]: dev isn't up right now - so i can't double check - but when i `print` the request - i get this

```
{
    'PV_API_VERSION_I': '',
    'PV_AD_ID_I': '',
    'PTYP_TP_TAB_I': '{"Data": [{"TP_CODE": "EL"}]}'
}
```

and the example in the spreadsheet looks like this

```
{
    "PV_API_VERSION_I": "1",
    "PV_AD_ID_I": "dugganpj@dosdev.us",
    "PTYP_TP_TAB_I": "{   \"Data\":{\"TP_CODE\": \"EL\"} }"
} 
```

i don't think this is correct - mainly cause the `[` after "Data" but also the escape characters (can't remember if they matter or not) going to need you to double check

**Review by M00staff — CHANGES_REQUESTED** [review]
going to need to double check the request format in swagger - also need confirmation that the overseas & domestic checkboxes on the frontend are not meant to be a part of the request

File: talentmap_api/fsbid/services/positions.py (hunk 1 above)
Comment (pveland621) [reply talentmap_api/fsbid/services/positions.py]: printing the request yields funky results, they tripped me up for a while with the escape characters not showing correctly in console but when you copy paste into DEV1, the request works. i think the console is just automatically taking the escape characters out but i think they're still there ASCII wise

Comment (pveland621) [reply talentmap_api/fsbid/services/positions.py]: > this looks a lot like the request structure for the `post_access` data - we made a common function that i THINK could have been used here called `format_request_data_to_string`
> 
> not worth holding this up, as this looks to be working just fine, as is ... but, if you have any problems with it down the line or for next time - might be worth looking into

gdi

Comment (pveland621) [reply talentmap_api/fsbid/services/positions.py]: also `[` after `Data` is required for sending multiple filter values for one filter (e.g. 3 bureaus selected as filters must be sent in an array

Comment (M00staff) [reply talentmap_api/fsbid/services/positions.py]: it works! 🎉

**Pull request merged**